### PR TITLE
Fix openFile breaking changes

### PR DESCRIPTION
**Pull request by Amejia481**
Requires https://github.com/mozilla-mobile/android-components/pull/10139 first

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt [hunk 1]
```diff
@@ -249,10 +249,20 @@ class DownloadFragment : LibraryPageFragment<DownloadItem>(), UserInteractionHan
...
+            } else {
+                0L
+            }
             AbstractFetchDownloadService.openFile(
-                context = it,
-                contentType = item.contentType,
-                filePath = item.filePath
+                it.applicationContext,
+                DownloadState(
+                    item.id,
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt:260]: Thanks for catching this :)  
I will update the PR

**Comment by Amejia481** [comment]
@gabrielluong I updated the PR!

File: app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt [hunk 2]
```diff
@@ -13,16 +13,20 @@ import mozilla.components.lib.state.Store
  * Class representing a downloads entry
  * @property id Unique id of the download item
  * @property fileName File name of the download item
+ * @property url The full url to the content that should be downloaded.
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt:16]: This looks out of place?

File: app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt [hunk 3]
```diff
@@ -13,16 +13,20 @@ import mozilla.components.lib.state.Store
...
  * @property filePath Full path of the download item
  * @property size The size in bytes of the download item
- * @property contentType The type of file the download is
+ * @property size The size in bytes of the download item
+ * @property url The type of file the download is
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt:20]: Hm? This should be contentType

File: app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt [hunk 4]
```diff
@@ -249,10 +250,21 @@ class DownloadFragment : LibraryPageFragment<DownloadItem>(), UserInteractionHan
...
+            val contentLength = if (item.size.isNotEmpty()) {
+                item.size.toLong()
+            } else {
+                0L
+            }
             AbstractFetchDownloadService.openFile(
-                context = it,
-                contentType = item.contentType,
-                filePath = item.filePath
+                it.applicationContext,
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt:259]: Let's add the named parameter for this too

**Review by gabrielluong — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt (hunk 3 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt:19]: Fixed!

Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt:19]: The description is still contentType.

**Review by gabrielluong — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt [hunk 5]
```diff
@@ -16,13 +16,16 @@ import mozilla.components.lib.state.Store
...
+ * @property url The full url to the content that should be downloaded
+ * @property status The status that represents every state that a download can be in
  */
 data class DownloadItem(
     val id: String,
     val fileName: String?,
     val filePath: String,
     val size: String,
     val contentType: String?,
+    val url: String,
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt:28]: I think you intended to make url the second property looking at this now.

**Review by gabrielluong — APPROVED** [review]

File: app/src/test/java/org/mozilla/fenix/ext/ListTest.kt [hunk 6]
```diff
@@ -58,9 +82,9 @@ class ListTest {
         file2.createNewFile()
         file3.createNewFile()
 
-        val item1 = DownloadItem("71", "filepath.txt", filePath1, "71 Mb", "text/plain", DownloadState.Status.COMPLETED)
-        val item2 = DownloadItem("72", "filepath2.txt", filePath2, "71 Mb", "text/plain", DownloadState.Status.COMPLETED)
-        val item3 = DownloadItem("73", "filepath3.txt", filePath3, "71 Mb", "text/plain", DownloadState.Status.COMPLETED)
+        val item1 = DownloadItem("71", "url", "filepath.txt", filePath1, "71 Mb", "text/plain", DownloadState.Status.COMPLETED)
+        val item2 = DownloadItem("72", "url", "filepath2.txt", filePath2, "71 Mb", "text/plain", DownloadState.Status.COMPLETED)
+        val item3 = DownloadItem("73", "url", "filepath3.txt", filePath3, "71 Mb", "text/plain", DownloadState.Status.COMPLETED)
```
Comment (gabrielluong) [inline app/src/test/java/org/mozilla/fenix/ext/ListTest.kt:87]: You could've copied what you had above and get those named parameters to future proof yourself

File: app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt (hunk 5 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragmentStore.kt]: Addressed

File: app/src/test/java/org/mozilla/fenix/ext/ListTest.kt (hunk 6 above)
Comment (Amejia481) [reply app/src/test/java/org/mozilla/fenix/ext/ListTest.kt]: Addressed

**Pull request merged**